src/main/kotlin/com/manimdsl/frontend/SemanticAnalysis.kt

```diff
@@ -116,6 +136,47 @@ class SemanticAnalysis {
         }
     }
 
+    fun incompatibleOperatorTypeCheck(
+        operator: String,
+        opExpr: ExpressionNode,
+        currentSymbolTable: SymbolTableVisitor,
+        ctx: ManimParser.ExprContext
+    ) {
+        when (opExpr) {
+            is BinaryExpression -> {
```

Comment:
A lot of repetition here - could the `incompatibleOperatorType` function not have an optional argument and then we have all of this code once?

Reply:
It can't as we need to know whether it is a binary or unary operator to get the expr1, expr2, expr members